app/code/Magento/Widget/Block/Adminhtml/Widget/Options.php

```diff
@@ -158,6 +159,10 @@ protected function _addField($parameter)
             $data['value'] = $parameter->getValue();
         }
 
+        if ($parameter->getType() == 'text' && $data['value'] != '') {
```

Comment:
To avoid the ` @SuppressWarnings(PHPMD.ExcessiveMethodLength)` we can create a private method and extract the IFs to handle the `$data['value']`, what do you think?